master/buildbot/test/integration/test_stop_triggered_not_started.py

```diff
@@ -0,0 +1,97 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+
+from buildbot.test.util.integration import RunMasterBase
+from buildbot.status.results import CANCELLED
+
+# This integration test creates a master and slave environment,
+# with two builders and a trigger step linking them. the triggered build never ends
+# so that we can reliabily stop it recursively
+
+
+class TriggeringMaster(RunMasterBase):
```

Comment:
I would prefer you try to factorize those two tests in one. (maybe do a reconfig for the second test)